### PR TITLE
Add tag functionality and delete confirmation

### LINKED ISSUE #17: As a user, I can list tasks by tag
...so that I can quickly filter for relevant tasks

### LINKED ISSUE #28: As a user I can create tags for tasks
... so that I can categorise them more easily (e.g. tutorial, project, assignment, etc)

### LINKED ISSUE #85: Add delete confirmation if user tries to delete a module with tasks in it

### LINKED ISSUE #98: Reset command not mentioned in in-program help description
The "reset" command word isn't mentioned under the list of available commands in the generic help message, although "help reset" does produce valid information on the "reset" command.

### PR DESCRIPTION
**Pull request by ngys117**
Added ability to add custom tags to tasks.

### CODE REVIEW

**Comment by ngys117** [comment]
Closes #17, #28, #85
Fixes #98
Maybe overlaps with #38

File: src/main/java/seedu/duke/parsers/TagParser.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+public class TagParser extends Parser {
+    public static final String TAG_OPERATION = StringConstants.TAG_OPERATION;
+    public static final String TASK_NUMBER = StringConstants.TASK_NUMBER;
+    public static final String TASK_MODULE = StringConstants.TASK_MODULE;
+    public static final String TAG_DESCRIPTION = StringConstants.TAG_DESCRIPTION;
+
+    //Unescaped Regex for testing:
+    //((?<taskOperation>\w+))(\s+(?<taskNumber>\d+))((\s+-m\s+(?<taskModule>\w+))?)(\s+(?<tagDescription>.+))
+    private static final String TAG_FORMAT = "((?<tagOperation>\\w+))(\\s+(?<taskNumber>\\d+))"
```
Comment (heekit73098) [inline src/main/java/seedu/duke/parsers/TagParser.java]: Perhaps you can restrict the tagOperation group to add or del to simplify where the exception is thrown

File: src/main/java/seedu/duke/parsers/TagParser.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            + "((\\s+-m\\s+(?<taskModule>\\w+))?)(\\s+(?<tagDescription>.+))";
```
Comment (heekit73098) [inline src/main/java/seedu/duke/parsers/TagParser.java]: if the tagDescription can contain spaces, maybe consider wrapping with double quotes for consistency with the other commands

File: src/main/java/seedu/duke/parsers/TagParser.java (hunk 1 above)
Comment (heekit73098) [inline src/main/java/seedu/duke/parsers/TagParser.java]: i think the regex might accept 
tag add 1 -m cs2113t
with the task description being "-m cs2113t", tested online so im not sure if it works as intended or not

**Review by heekit73098 — COMMENTED** [review]
Implementation generally good, just a few comments on the regex and command format

File: src/main/java/seedu/duke/parsers/TagParser.java (hunk 1 above)
Comment (ngys117) [reply src/main/java/seedu/duke/parsers/TagParser.java]: Good catch, updated the regex.

File: src/main/java/seedu/duke/parsers/TagParser.java (hunk 2 above)
Comment (ngys117) [reply src/main/java/seedu/duke/parsers/TagParser.java]: Updated the regex to accept only one word wrapped around the quotes, no white spaces allowed.

**Pull request merged**